lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -0,0 +1,116 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+
+-- |
+-- Copyright: © 2020 IOHK
+-- License: Apache-2.0
+--
+-- Working with Shelley transactions.
+
+module Cardano.Wallet.Shelley.Transaction
+    ( newTransactionLayer
+    , fromGenesisTxOut
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( NetworkDiscriminant (..) )
+import Cardano.Wallet.Primitive.Types
+    ( Address (..)
+    , Block (..)
+    , BlockHeader (..)
+    , BlockchainParameters (..)
+    , Hash (..)
+    , ProtocolMagic (..)
+    , SlotId (..)
+    , Tx (..)
+    , TxOut (..)
+    )
+import Cardano.Wallet.Shelley.Compatibility
+    ( Shelley )
+import Cardano.Wallet.Transaction
+    ( ErrValidateSelection, TransactionLayer (..) )
+import Crypto.Hash
+    ( hash )
+import Crypto.Hash.Algorithms
+    ( Blake2b_256 )
+import Data.ByteString
+    ( ByteString )
+import Data.Coerce
+    ( coerce )
+import Data.Proxy
+    ( Proxy )
+import Data.Quantity
+    ( Quantity (..) )
+import Fmt
+    ( Buildable (..) )
+import GHC.Stack
+    ( HasCallStack )
+
+import qualified Data.ByteArray as BA
+import qualified Data.ByteString as BS
+
+
+newTransactionLayer
+    :: forall (n :: NetworkDiscriminant) k t c.
+        ( t ~ IO (Shelley c))
+    => Proxy n
+    -> ProtocolMagic
+    -> TransactionLayer t k
+newTransactionLayer _proxy _protocolMagic = TransactionLayer
+    { mkStdTx = notImplemented "mkStdTx"
+    , mkDelegationJoinTx = notImplemented "mkDelegationJoinTx"
+    , mkDelegationQuitTx = notImplemented "mkDelegationQuitTx"
+    , decodeSignedTx = notImplemented "decodeSignedTx"
+    , estimateSize = notImplemented "estimateSize"
+    , estimateMaxNumberOfInputs = notImplemented "estimateMaxNumberOfInputs"
+    , validateSelection = notImplemented "validateSelection"
+    , allowUnbalancedTx = notImplemented "allowUnbalancedTx"
+    }
+
+fromGenesisTxOut :: BlockchainParameters -> [TxOut] -> Block
+fromGenesisTxOut bp outs = Block
+    { delegations  = []
+    , header = BlockHeader
+        { slotId =
+            SlotId 0 0
+        , blockHeight =
+            Quantity 0
+        , headerHash =
+            coerce $ getGenesisBlockHash bp
+        , parentHeaderHash =
+            Hash (BS.replicate 32 0)
+        }
+    , transactions = mkTx <$> outs
+    }
+  where
+    -- TODO: Not sure why we need this function, but should we call blake2b256
+    -- for Shelley really?
+    mkTx out@(TxOut (Address bytes) _) =
+        Tx (Hash $ blake2b256 bytes) [] [out]
```

Comment:
This function is needed for our integration test setup. With Jörmungandr, the genesis file is mapped to a genesis block which contains a set of initial transaction (although with no inputs). On cardano-node however, the genesis contains an initial set of Genesis TxOut, so this function artificially maps that to an initial "Block". 

May you clarify this with a comment for future readers?